workflow/controller/workflowpod_test.go

```diff
@@ -77,6 +77,65 @@ func TestScriptTemplateWithVolume(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+var scriptTemplateWithOptionalInputArtifactProvided = `
+name: script-with-input-artifact
+inputs:
+  artifacts:
+  - name: manifest
+    path: /manifest
+    optional: true
+    http:
+        url: https://raw.githubusercontent.com/argoproj/argo/stable/manifests/install.yaml
+script:
+  image: alpine:latest
+  command: [sh]
+  source: |
+    ls -al
+`
+
+var scriptTemplateWithOptionalInputArtifactNotProvided = `
+name: script-with-input-artifact
+inputs:
+  artifacts:
+  - name: manifest
+    path: /manifest
+    optional: true
+script:
+  image: alpine:latest
+  command: [sh]
+  source: |
+    ls -al
```

Comment:
Tried to capture the result of this command and test on it but was not able to find a way to do it. Is there a way how?

Reply:
The latest implementation will ignore the script output if there is no reference for output in template.